core/modules/path/path.module

```diff
@@ -453,10 +808,875 @@ function path_entity_update(Entity $entity) {
+  //if (isset($_GET['destination'])) {
+    // Simply unset the parameter since it has already been passed into
+    // $options['query'] in the previous code.
+  //  unset($_GET['destination']);
+  //}
+
+  // Allow other modules to alter the redirect before passing to backdrop_goto().
+  backdrop_alter('redirect', $redirect);
+
+  // Continue if the redirect has not been disabled by hook_redirect_alter().
+  if (isset($redirect->redirect) && isset($redirect->callback) && $redirect->redirect !== FALSE && function_exists($redirect->callback)) {
+    // Perform the actual redirect.
+    $callback = $redirect->callback;
+    $callback($redirect);
+  }
+}
+
+/**
+ * Redirect callback; perform an URL redirect.
+ */
+function redirect_goto($redirect) {
+  $redirect->redirect_options['absolute'] = TRUE;
+  $url = url($redirect->redirect, $redirect->redirect_options);
+  backdrop_add_http_header('Location', $url);
+  backdrop_add_http_header('Status', redirect_status_code_options($redirect->status_code));
+
+  if (!empty($redirect->rid)) {
+    // Add a custom header for the redirect ID so when the redirect is served
+    // from the page cache, we can track it.
+    backdrop_add_http_header('X-Redirect-ID', $redirect->rid);
+  }
+  $config = config('redirect.settings');
+  $sys_config = config('system.core');
+  if (!$config->get('page_cache') || !$sys_config->get('cache') || !backdrop_page_is_cacheable() || empty($redirect->cache)) {
+    backdrop_exit($url);
+  }
+
+  // @see backdrop_exit()
+  if (backdrop_get_bootstrap_phase() == BACKDROP_BOOTSTRAP_FULL) {
+    if (!defined('MAINTENANCE_MODE') || MAINTENANCE_MODE != 'update') {
+      module_invoke_all('exit', $url);
+    }
+    backdrop_session_commit();
+    if ($sys_config->get('cache')) {
+      // We must output something to allow the request to be cached.
+      echo ' ';
+      if ($cache = backdrop_page_set_cache()) {
+        // When caching this redirect for the first time we still need to ensure
+        // that the correct cache headers are sent.
+        // @see backdrop_page_footer()
+        backdrop_serve_page_from_cache($cache);
+      }
+    }
+  }
+
+  exit;
+}
+
+/**
+ * Clear a page from the page cache.
+ */
+function redirect_page_cache_clear($redirect = NULL) {
+  $config = config('redirect.settings');
+  if (!$config->get('page_cache')) {
+    return;
+  }
+
+  if (isset($redirect)) {
+    $path = url($redirect->source, array('absolute' => TRUE));
+    // Use a wildcard to catch paths with query strings.
+    cache_clear_all($path, 'cache_page', TRUE);
+  }
+  else {
+    // Clear the entire page cache.
+    cache_clear_all('*', 'cache_page', TRUE);
+  }
+}
+
+/**
+ * Given a path determine if it is an entity default path.
+ *
+ * @param $path
+ *   The internal path. The id of the entity should be in the string as '[id]'.
+ * @return
+ *   An array with the entity type and the loaded entity object.
+ */
+function redirect_load_entity_from_path($path) {
```

Comment:
It doesn't look like this function would work in our current code base, but it is only called from commented out code above. Maybe remove until we need it.